adding a subset of cis benchmarks policies to canary workstations team

dherder opened the pull request.

dherder reviewed (commented):
5 policies added per OS for CIS benchmarks

noahtalerman commented:
Hey @dherder the policies you added are using the `fleetctl apply` format instead of the best practice GitOps format.

This means the policies won't get added to Fleet. See the error in the "Apply latest configuration to dogfood test"

The best practice GitOps format looks like this:

```yml
- name: macOS - Enable FileVault
  query: SELECT 1 FROM filevault_status WHERE status = 'FileVault is On.';
  critical: false
  description: This policy checks if FileVault (disk encryption) is enabled.
  resolution: As an IT admin, turn on disk encryption in Fleet.
  platform: darwin
```

We removed `apiVersion`, `kind`, and `spec` keys because they're redundant. They don't change.

See more examples in the `macos-device-health.policies.yml here: https://github.com/fleetdm/fleet/blob/main/it-and-security/lib/macos-device-health.policies.yml

dherder commented:
> Hey @dherder the policies you added are using the `fleetctl apply` format instead of the best practice GitOps format.
> 
> This means the policies won't get added to Fleet. See the error in the "Apply latest configuration to dogfood test"
> 
> The best practice GitOps format looks like this:
> 
> ```yaml
> - name: macOS - Enable FileVault
>   query: SELECT 1 FROM filevault_status WHERE status = 'FileVault is On.';
>   critical: false
>   description: This policy checks if FileVault (disk encryption) is enabled.
>   resolution: As an IT admin, turn on disk encryption in Fleet.
>   platform: darwin
> ```
> 
> We removed `apiVersion`, `kind`, and `spec` keys because they're redundant. They don't change.
> 
> See more examples in the `macos-device-health.policies.yml here: https://github.com/fleetdm/fleet/blob/main/it-and-security/lib/macos-device-health.policies.yml

@noahtalerman Ah, I just extracted a portion of the policies from the CIS benchmarks .yml. Do we need to publish 2 versions of the benchmark policies or should we change the format of the published version to be GitOps compatible?

noahtalerman commented:
> Do we need to publish 2 versions of the benchmark policies or should we change the format of the published version to be GitOps compatible?

@dherder I think we should change the best practice version to be GitOps compatible but maintain the `fleetctl apply` format for backward compatibility.

There's an issue tracking this here but it hasn't been prioritized: https://github.com/fleetdm/fleet/issues/17913

dherder commented:
@noahtalerman totally get it. In order to unblock @phtardif1 I altered the format of the yml to be gitops compatible in the latest commit. Can you review it please?

lukeheath reviewed (changes requested):
Please move this to the "Workstations" team instead of "Workstations (Canary)". 

We are currently using canary to dogfood and demo the calendar flow including OS updates, so we don't want the OS to handle updates in canary. 

cc @spokanemac

lukeheath commented:
@dherder Looks good but has a conflict.
